src/azul/chalice.py

```diff
@@ -467,6 +476,40 @@ def swagger_resource(self, file) -> Response:
                                 headers={'Content-Type': content_type},
                                 body=body)
 
+    def threshold(self, *, errors: int, throttles: int):
+        def wrapper(f):
+            f.errors_threshold = errors
+            f.throttles_threshold = throttles
+            return f
+
+        return wrapper
+
+    def metric_thresholds(self) -> list[MetricThreshold]:
+        default_threshold = 0
+        metrics = ('errors', 'throttles')
+        thresholds = []
+        lambda_name, _ = config.unqualified_resource_name(self.app_name)
+        for metric in metrics:
+            # The api_handler lambda functions (indexer & service) aren't included
+            # in the app_module's handler_map, so we account for those first.
+            threshold = getattr(self, f'{metric}_threshold', default_threshold)
+            thresholds.append(MetricThreshold(lambda_name=lambda_name,
+                                              handler_name=None,
+                                              metric=metric,
+                                              threshold=threshold))
+            for handler in self.handler_map.values():
+                if isinstance(handler, chalice.app.EventSourceHandler):
+                    if isinstance(handler.func, chalice.app.PureLambdaWrapper):
+                        func = handler.func._original_func
+                    else:
+                        func = handler.func
```

Comment:
A bit more concise. This is a nitpick, feel free to ignore.
```suggestion
                    func = handler.func
                    if isinstance(handler.func, chalice.app.PureLambdaWrapper):
                        func = func._original_func                       
```